### ISSUE TITLE
'ascii' codec can't encode character u'\u25cf' during SSHClient.run

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1096]
When performing a SSHClient.run call I encountered this problem on one of our 16.04 CI environments:

```
In [7]: client = SSHClient('10.100.199.182', username='root')

In [8]: client.run('systemctl status ovs-arakoon-ovsdb.service')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-8-e36dac058b7f> in <module>()
----> 1 client.run('systemctl status ovs-arakoon-ovsdb.service')

/opt/OpenvStorage/ovs/extensions/generic/sshclient.pyc in inner_function(self, *args, **kwargs)
     57                 if self._client is not None and not self._client.is_connected():
     58                     self._connect()
---> 59                 return outer_function(self, *args, **kwargs)
     60             except AttributeError as ex:
     61                 if "'NoneType' object has no attribute 'open_session'" in str(ex):

/opt/OpenvStorage/ovs/extensions/generic/sshclient.pyc in run(self, command, debug, suppress_logging)
    265                 command = ' '.join(command)
    266             _, stdout, stderr = self._client.exec_command(command)  # stdin, stdout, stderr
--> 267             output = self._clean_text(stdout.readlines())
    268             error = self._clean_text(stderr.readlines())
    269             exit_code = stdout.channel.recv_exit_status()

/opt/OpenvStorage/ovs/extensions/generic/sshclient.pyc in _clean_text(text)
    208         # This strip is absolutely necessary. Without it, channel.communicate() is never executed (odd but true)
    209         try:
--> 210             cleaned = text.strip().decode('utf-8', 'replace')
    211             for old, new in {u'\u2018': "'",
    212                              u'\u201a': "'",

/usr/lib/python2.7/encodings/utf_8.pyc in decode(input, errors)
     14 
     15 def decode(input, errors='strict'):
---> 16     return codecs.utf_8_decode(input, errors, True)
     17 
     18 class IncrementalEncoder(codecs.IncrementalEncoder):

UnicodeEncodeError: 'ascii' codec can't encode character u'\u25cf' in position 0: ordinal not in range(128)
```

The exception is probably referring to this ascii icon: http://www.fileformat.info/info/unicode/char/25CF/index.htm

**Comment by khenderick** [comment]
> When you call .decode() an a unicode string, Python 2 tries to be helpful and decides to encode the Unicode string back to bytes (using the default encoding), so that you have something that you can really decode. This implicit encoding step doesn't use errors='replace', so if there are any characters in the Unicode string that aren't in the default encoding (probably ASCII) you'll get a UnicodeEncodeError.

Python, y u so annoying...

**Comment by khenderick** [comment]
Fixed by #1110, packaged in openvstorage-2.7.4-rev.4246.ac6497b

**Comment by kinvaris** [comment]
PASSED: We did not encounter this anymore on CI environments
